### ISSUE TITLE
Why model strategies have larger program sizes instead of smaller ones

### ISSUE THREAD

**Issue by 18liumin** [google/ml-compiler-opt#331]
use -enable-ml-inliner=release
cmake  -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=${PWD}/install -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DTEST_SUITE_COLLECT_COMPILE_TIME=OFF -DCMAKE_C_FLAGS="'-fuse-ld=lld   -static -mllvm  -enable-ml-inliner=release'" -DCMAKE_CXX_FLAGS="'-fuse-ld=lld - -static  -mllvm -enable-ml-inliner=release'"  -DTEST_SUITE_SPE
C2017_ROOT=/usr1/project/cpu2017 -C ../test-suite/cmake/caches/Oz.cmake /usr1/project/llvm-install/test-suite

no -enable-ml-inliner=release
cmake  -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=${PWD}/install -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DTEST_SUITE_COLLECT_COMPILE_TIME=OFF -DCMAKE_C_FLAGS="'-fuse-ld=lld --target=aarch64-linux-gnu  -static -Xc
lang -fembed-bitcode=all'" -DCMAKE_CXX_FLAGS="'-fuse-ld=lld   -static  l'"  -DTEST_SUITE_SPEC2017_ROOT=/usr1/project/cpu2017 -C ../test-suite/cmake/caches/Oz.cmake /usr1/project/llvm-install/test-suite

(base) @szvphis6315984:~/project/llvm-install$  llvm-size test-build-model/602.gcc_s
   text    data     bss     dec     hex filename
8439896  458417 1170152 10068465         99a1f1 test-build-model/602.gcc_s
(base) @szvphis6315984:~/project/llvm-install$ llvm-size test-build-no-model/602.gcc_s
   text    data     bss     dec     hex filename
8360512  458377 1170152 9989041  986bb1 test-build-no-model/602.gcc_s

(base) @szvphis6315984:~/project/llvm-install$ llvm-size test-build-model/526.blender_r
   text    data     bss     dec     hex filename
11232454        4918192  450328 16600974         fd4f8e test-build-model/526.blender_r
(base) @szvphis6315984:~/project/llvm-install$ llvm-size test-build-no-model/526.blender_r
   text    data     bss     dec     hex filename
11101990        4917904  450296 16470190         fb50ae test-build-no-model/526.blender_r

**Comment by mtrofin** [comment]
Most probably the model wasn't one trained on a corpus representative to what you're validating on - what model did you use? See also Issue #310, [this comment](https://github.com/google/ml-compiler-opt/issues/310#issuecomment-1772785386).

If you're using the latest model we released, that was trained on predominantly C++ systems/server-side code, which looks nothing like spec. Works well on clang though.

About `-static` - not sure; so the absolute difference between base and experiment changes when you pass `-static`?

**Comment by 18liumin** [comment]
> Most probably the model wasn't one trained on a corpus representative to what you're validating on - what model did you use? See also Issue #310, [this comment](https://github.com/google/ml-compiler-opt/issues/310#issuecomment-1772785386).
> 
> If you're using the latest model we released, that was trained on predominantly C++ systems/server-side code, which looks nothing like spec. Works well on clang though.
> 
> About `-static` - not sure; so the absolute difference between base and experiment changes when you pass `-static`?

yes, With static linking, the model results will be worse. With dynamic linking, the model results will be better
